client/src/pages/Search/Search.js

```diff
@@ -0,0 +1,88 @@
+import React, { useState } from 'react';
+import { useQuery } from 'react-query';
+import { debounce } from '@mui/material/utils';
+import SearchBar from '@/components/Search/SearchBar';
+import { mapboxAccessToken } from '@/util/config';
+
+import './Search.scss';
+
+const MIN_CHARS_FOR_SEARCH = 2;
+const BASE_MAPBOX_SEARCH_API_URL =
+  'https://api.mapbox.com/geocoding/v5/mapbox.places/';
+const SEARCH_QUERY_CACHE_TIME = 1000 * 30; // 30 seconds
+
+const Search = ({ map }) => {
+  const [query, setQuery] = useState('');
+  const {
+    data: searchResults,
+    isLoading,
+    isError,
+    isSuccess,
+  } = useQuery({
+    queryKey: query,
+    queryFn: () => getSearchResults(query),
+    enabled: query.length > MIN_CHARS_FOR_SEARCH,
+    cacheTime: SEARCH_QUERY_CACHE_TIME,
+  });
+
+  // Debounce search reqeusts to mitigate churning through our API requests budget
+  const debouncedSetQuery = debounce((query) => setQuery(query), 250);
+  const handleInputChange = (event) => {
+    debouncedSetQuery(event.currentTarget.value);
+  };
+
+  const handleOptionSelect = (e, option) => {
+    if (option) {
+      map.panTo(option.coords);
+    }
+  };
+
+  let options = [];
+  if (searchResults?.features.length) {
+    options = searchResults.features.map((result) => ({
+      label: result.text,
+      address: result.place_name,
+      id: result.id,
+      coords: result.center,
+      type: 'Results',
+    }));
+  } else if (isSuccess) {
+    options = [{ label: 'No results found' }];
+  } else if (isError) {
+    options = [{ label: 'Error encountered. Please try again later.' }];
+  }
+
+  return (
+    <div className="search-container">
+      <SearchBar
+        options={options}
+        loading={isLoading}
+        handleInputChange={handleInputChange}
+        handleOptionSelect={handleOptionSelect}
+      />
+    </div>
+  );
+};
+
+const getSearchResults = async (query) => {
+  const hashParams = new URLSearchParams(window.location.hash.slice(1));
+  const coords = hashParams.get('pos')?.split('/');
+  const searchParams = new URLSearchParams({
+    access_token: mapboxAccessToken,
+    autocomplete: true,
+    fuzzymatch: true,
+    proximity: `${coords[2]},${coords[1]}`,
```

Comment:
This biases the search towards results close to where the map is currently centered. It's actually pretty cool and I plan to make this smarter in future work. 

Tldr; this proximity makes address results so much more accurate when I'm looking at somewhere on the map that is close by my search query. It actually makes it less accurate when searching after initial zoomed out load since it searches for results around middle America